src/node.cc

```diff
@@ -994,6 +984,14 @@ InitializationResult InitializeOncePerProcess(int argc, char** argv) {
     }
   }
 
+  if (per_process::cli_options->use_largepages) {
+    if (node::IsLargePagesEnabled()) {
+      if (node::MapStaticCodeToLargePages() != 0) {
+        fprintf(stderr, "Reverting to default page size\n");
```

Comment:
should we confine this printf under debug builds only?

Reply:
There should be some feedback though if it fails to map, right? Like, there are no performance gains because it failed to map to large pages, not because it succeeded, but the application is such that it doesn't help.

Reply:
@gireeshpunathil also, how do I confine it to debug builds?

```C++
#ifdef DEBUG
fprintf(...);
#endif  // DEBUG
```
?

Reply:
I think it is better to place it in the documentation to state that the option is silently ignored if the OS is not configured to support that.

> #ifdef DEBUG

yes.

Reply:
I think we generally don't shy away from printing to stderr when a command line option does not work (e.g. warnings are printed when the v8 coverage or profiles cannot be written to disk). For consistency I'd prefer we print to stderr without a DEBUG guard as well.

Reply:
Just to +1 that, I'd rather see this changed back again to the simple boolean option.

Reply:
in terms of consistency, there are many optimizers / accelerators across jit and gc, that silently pick up the next viable approach when their best bet fails, but do not print out what is happening underneath and what decisions they made. the subject of current PR is an optimization, and imo a runtime should not be verbose around such capabiities, which fits more for a parser or an installer use cases.

On the other hand, I believe the documentation should be the key - following the `madvise`'s example itself, the doc should be explicit in terms of the option as liable to fail because of reasons outside the control of user's and node's scope; and is a directive, not a command. Or else, the abstraction becomes loose with many potential outcomes and side effects.